plum/method.py

```diff
@@ -77,13 +82,131 @@ def __hash__(self):
+        if sig.precedence != Signature._default_precedence:
+            res.append(f"\n    precedence={sig.precedence}")
+
+        yield res
+        yield Segment("    ")
+        yield repr_pyfunction(self.implementation)
+
+    def _repr_signature_mismatch(self, args_ok: List[bool]) -> str:
+        """Special version of __repr__ but used when
+        printing args mismatch (mainly in hints to possible
+        similar signatures).
+
+        Args:
+            args_ok: a list of which arguments match this signature
+                and which don't according to the resolver.
+        """
+        sig = self.signature
+
+        argnames, kwnames, kwvar_name = extract_argnames(self.implementation)
+        varargs_ok = all(args_ok[len(sig.types) :])
+
+        parts = []
+        if sig.types:
+            for i, (nm, t) in enumerate(zip(argnames, sig.types)):
+                is_ok = args_ok[i] if i < len(args_ok) else False
+                arg_txt = Text(f"{nm}: ")
+                type_txt = repr_type(t)
+                if not is_ok:
+                    type_txt.stylize("red")
+                arg_txt.append(type_txt)
+                parts.append(arg_txt)
+        if sig.varargs != Signature._default_varargs:
+            arg_txt = Text(f"*{argnames[-1]}: ")
+            type_txt = repr_type(sig.varargs)
+            if not varargs_ok:
+                type_txt.stylize("red")
+            arg_txt.append(type_txt)
+            parts.append(arg_txt)
+
+        if len(kwnames) > 0 or kwvar_name is not None:
+            parts.append(Text("*"))
+        for kwnm in kwnames:
+            parts.append(Text(f"{kwnm}"))
+        if kwvar_name is not None:
+            parts.append(Text(f"**{kwvar_name}"))
+
+        res = Text(self.function_name) + Text("(") + Text(", ").join(parts) + Text(")")
+        if self.return_type != Method._default_return_type:
+            res.append(" -> ")
+            res.append(repr_type(self.return_type))
+        if sig.precedence != Signature._default_precedence:
+            res.append(f"\n    precedence={sig.precedence}")
+
+        res.append("\n    ")
+        res.append_text(repr_pyfunction(self.implementation))
+        return res
+
+
+@rich_repr
+class MethodList(list):
+    def __rich_console__(self, console, options):
+        yield f"Method List with # {len(self)} methods:"
+        for i, method in enumerate(self):
+            yield Segment(f" [{i}] ")
+            yield method
+
+
+def extract_argnames(f: Callable, precedence: int = 0) -> Signature:
+    """Extract the signature from a function.
+
+    Args:
+        f (function): Function to extract signature from.
+        precedence (int, optional): Precedence of the method.
+
+    Returns:
+        :class:`.Signature`: Signature.
+    """
+    # Extract specification.
+    sig = inspect_signature(f)
+
+    # Get types of arguments.
+    argnames = []
+    kwnames = []
+    kwvar_name = None
```

Comment:
```suggestion
    regular_args = []
    kw_only_args = []
    var_kw_name = None
```

where `var_kw_name` stands for `variable_keykword_name`.

Would u be ok with these variable names? Trying to make things as clear as possible.